sklearn/ensemble/_hist_gradient_boosting/_predictor.pyx

```diff
@@ -218,7 +219,8 @@ def _compute_partial_dependence(
                 # determine if the split feature is a target feature
                 is_target_feature = False
                 for feature_idx in range(target_features.shape[0]):
-                    if target_features[feature_idx] == current_node.feature_idx:
+                    target_feature_of_feature_idx = target_features[feature_idx]
+                    if target_feature_of_feature_idx == current_node.feature_idx:
```

Comment:
instead of this, I think this should be changed in the first place:

```
int [:] target_features
# to
unsigned int [:] target_features
```

and all references and usages to `target_features` as well.

Reply:
good point, changed it.

Reply:
Note that you'll also now need to ensure it is only ever provided `unsigned int` data. That should fix the failing tests. Also, should these not be of type `intp_t` from `_typedefs.pxd` if they're being used as indices?